### PR TITLE
Lint fixes

### PR DESCRIPTION
**Pull request by locochris**
Fixes for the failing PR #66

### CODE REVIEW

**Comment by rylnd** [comment]
This is awesome. Thank you for figuring this out!

**Comment by rylnd** [comment]
@locochris can you rebase this branch? I'd like those first two commits squashed, and @hlangeveld's commit removed. 

In general, let's try to branch from master unless another branch is absolutely necessary; I've gone down that road, and it's messy and time-consuming. In this case, that would have meant just allowing linting failures until @hlangeveld's branch was merged.

**Comment by locochris** [comment]
@rylnd - righto - so this branch is now just the fixes suggested by the linter and #66 is adding the `shellcheck` linter.  The @hlangeveld's branch should fix the remaining `echoe` failure.  And we can add `sh` in another commit/branch.

**Comment by rylnd** [comment]
Cool, I'm gonna merge this and then #66, if that makes sense.

File: bin/shpec [hunk 1]
```diff
@@ -22,7 +22,7 @@ describe() {
 end() {
   : $((test_indent -= 1))
   if [ $test_indent -eq 0 ]; then
-    [ $failures -eq 0 ]
+    [ "$failures" -eq 0 ]
```
Comment (rylnd) [inline bin/shpec]: @locochris this is a weird one; why would the linter complain about this line but not the one above it?

Comment (locochris) [inline bin/shpec]: @rylnd - yer no idea - and the same line occurs in `final_results` - so i don't have 100% confidence in this linter, but at least it has *some* useful suggestions.

Comment (rylnd) [inline bin/shpec]: Welp, that line is gonna be removed in #71 anyway, so moot point :wink:

**Pull request merged**